File: BringMeHome/BringMeHome.Models/SearchObjects/AnimalSearchObject.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BringMeHome.Models.SearchObjects
{
    public class AnimalSearchObject:BaseSearchObject
    {
        public int? ColorID { get; set; }
        public int? TemperamentID { get; set; }
        public int? SpeciesID { get; set; }
        public string? SpeciesName { get; set; }
        public int? BreedID { get; set; }
        public string? BreedName { get; set; }
        public int? ShelterID { get; set; }
        public int? CantonID { get; set; }
        public string? Status { get; set; }
        public bool? AvailableOnly { get; set; }
    }
}

File: bringmehomeUI/bringmehome_android/lib/models/search_objects/animal_search_object.dart
class AnimalSearchObject {
  String? fts;
  int? colorID;
  int? temperamentID;
  int? speciesID;
  int? breedID;
  int? cantonID;
  int? shelterID;
  int pageNumber;
  int pageSize;

  AnimalSearchObject({
    this.fts,
    this.colorID,
    this.temperamentID,
    this.speciesID,
    this.breedID,
    this.cantonID,
    this.shelterID,
    this.pageNumber = 1,  
    this.pageSize = 10,  
  });

  Map<String, dynamic> toJson() {
    return {
      if (fts != null) 'FTS': fts,
      if (colorID != null) 'ColorID': colorID,
      if (temperamentID != null) 'TemperamentID': temperamentID,
      if (speciesID != null) 'SpeciesID': speciesID,
      if (breedID != null) 'BreedID': breedID,
      if (cantonID != null) 'CantonID': cantonID,
      if (shelterID != null) 'ShelterID': shelterID,
      'PageNumber': pageNumber,
      'PageSize': pageSize,
    };
  }
}